### PR TITLE
[Revised] iOS ReplayKit Streaming

### LINKED ISSUE #661: [Deferred] SDL 0215 - iOS ReplayKit Streaming
Hello SDL community,

The review of "SDL 0215 - iOS ReplayKit Streaming" begins now and runs through February 19, 2019. The proposal is available here:

https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0215-ios-replaykit-streaming.md

Reviews are an important part of the SDL evolution process. All reviews should be sent to the associated Github issue at:

https://github.com/smartdevicelink/sdl_evolution/issues/661

What goes into a review?

The goal of the review process is to improve the proposal under review through constructive criticism and, eventually, determine the direction of SDL. When writing your review, here are some questions you might want to answer in your review:

* Is the problem being addressed significant enough to warrant a change to SDL?
* Does this proposal fit well with the feel and direction of SDL?
* If you have used competitors with a similar feature, how do you feel that this proposal compares to those?
* How much effort did you put into your review? A glance, a quick reading, or an in-depth study?
Please state explicitly whether you believe that the proposal should be accepted into SDL.

More information about the SDL evolution process is available at

https://github.com/smartdevicelink/sdl_evolution/blob/master/process.md

Thank you,
Jordyn Mackool

### PR DESCRIPTION
**Pull request by SatbirTanda**
This proposal aims to optimize how the car window API captures the screen in the iOS Framework using ReplayKit.

Fixes #661

### CODE REVIEW

File: proposals/0215-ios-replaykit-streaming.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+* Nothing beyond keeping the original design.
+
+## Update
+
+* After testing the SDL Lock Screen appearance, ReplayKit does not take in a root veiw controller to display it seems to record the current window that is displayed.  Because the SDL Lockscreen shows up on another window, it does not seem to impede the recording of the original window.
+* Attached in proposal issue is an app that can be run as an example - there still needs to be code that resizes the pixel buffer (need to see if the is possible at all, the snippet provided did not alter the stream)
+* The app does not need to be landscaped if we can solve the pixel sizing issue, however, then we would also need to make sure the rootview controller does not resize on video streaming 
+* I determined the replay kit stream also contains audio data that can used to stream audio to the Head Unit exactly like a media app.  This would mean the developer would no longer need to send PCM audio files to SDL (they can play audio as they would in their app and it should carry over).
+* There would need tests to be done with iPhone XS, X, XS Max, XR - I am seeing a crash on these phones.
```
Comment (jordynmackool) [inline proposals/0215-ios-replaykit-streaming.md]: Update and please refrain from using "I":
* Tests would need to be done with iPhone XS, X, XS Max, XR - when testing these phones, there has been notable crashes.

**Comment by jordynmackool** [comment]
@SatbirTanda is this ready for re-review?

**Comment by SatbirTanda** [comment]
@jordynmackool Yes please re-review

File: proposals/0215-ios-replaykit-streaming.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+* No changes to the Android library
+* The iOS library would need to alter it's code to add a flag to pick between ReplayKit streaming or the current method of streaming (possibly via the ```SDLStreamingMediaConfiguration``` ).  Additionally, it would need to alter SDLCarWindow.m's ```syncFrame``` method that causes the jump in CPU usage (which can be seen via the CPU profiling instrument while running an SDL enabled navigation app by filtering out system methods).  The app would need to be forced in a horizontal orientation.
+
+## Alternatives considered
+
+* There are private methods that can capture the screen very efficiently, however, implementing these can get your app removed from the app store or rejected from submission by Apple.
+
+* Nothing beyond keeping the original design.
+
+## Update
```
Comment (jordynmackool) [inline proposals/0215-ios-replaykit-streaming.md]: Can you please alter the title`Updates` to `Amendments`

**Pull request merged**